src/main/java/com/google/sps/servlets/ChatServlet.java

```diff
@@ -0,0 +1,103 @@
+package com.google.sps.servlets;
+
+import com.google.sps.data.Message;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.EntityNotFoundException;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.gson.Gson;
+import java.util.LinkedList;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+@WebServlet("/chat")
+
+public class ChatServlet extends HttpServlet {
+
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+      long roomID = Long.parseLong(request.getQueryString());
+      DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    
+    Gson gson = new Gson();
+
+     // Handle entity not found exception
+    try{
+        // Retrieves the entity with matching ID and its corresponding messages property as a JSON string
+        Entity roomEntity = datastore.get(KeyFactory.createKey("Room", roomID));
+        String jsonMessages = (String) roomEntity.getProperty("messages");
```

Comment:
Does this produce a json message when stored in datastore?

Also, roomEntity should be retrieved from a new function called Room.getRoomFromId(roomId) which queries datastore and converts to Room object.